.github/workflows/ubuntu.yml

```diff
@@ -0,0 +1,42 @@
+name: Ubuntu 18.04 Tests
+
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+
+# This allows a subsequently queued workflow run to interrupt previous runs.
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  ubuntu_ci:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Adding HHVM repo
+        run: |
+          curl -fsSL "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x058341c68fc8de6017d775a1b4112585d386eb94" | gpg --dearmor > hhvm-archive-keyring.gpg
+          sudo mv hhvm-archive-keyring.gpg /usr/share/keyrings
+          echo "deb [arch=amd64 signed-by=/usr/share/keyrings/hhvm-archive-keyring.gpg] https://dl.hhvm.com/ubuntu bionic main" | sudo tee /etc/apt/sources.list.d/hhvm.list
```

Comment:
This is based on the instructions in the HHVM website: https://docs.hhvm.com/hhvm/installation/building-from-source#installing-build-dependencies__debian-or-ubuntu

Instead of using `add-key` and `add-apt-repository`, I fetch the key directly like Bazel as this will work on Ubuntu 22.04.